[CI] TickerScheduleEngineTests testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval failing

**Build Scans:**
- [elasticsearch-intake #11747 / part2](https://gradle-enterprise.elastic.co/s/crk6vmtwkmcj6)
- [elasticsearch-pull-request #38035 / part-2](https://gradle-enterprise.elastic.co/s/thxsgrxyujgte)
- [elasticsearch-pull-request #38029 / part-2](https://gradle-enterprise.elastic.co/s/5haqmmv3d4gw6)
- [elasticsearch-pull-request #38025 / part-2](https://gradle-enterprise.elastic.co/s/tggqml4kfdmyw)
- [elasticsearch-pull-request #38000 / part-2](https://gradle-enterprise.elastic.co/s/koagjcpc4wvpm)
- [elasticsearch-periodic #4544 / openjdk23_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/qdxqbka3d3zde)
- [elasticsearch-pull-request #37883 / part-2](https://gradle-enterprise.elastic.co/s/osl6ux275rmxu)
- [elasticsearch-periodic-platform-support #4511 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/kdlromn7az46k)
- [elasticsearch-periodic #4539 / openjdk23_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/lwxfz5sukxul4)
- [elasticsearch-pull-request #37808 / part-2](https://gradle-enterprise.elastic.co/s/uqslsbae65x6a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:watcher:test" --tests "org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests.testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval" -Dtests.seed=3FF51600300F454D -Dtests.locale=ar-BH -Dtests.timezone=America/Rio_Branco -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <2>
     but: was <1>
```

**Issue Reasons:**
- [8.x] 3 consecutive failures in test testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval
- [8.x] 2 consecutive failures in step part-2
- [8.x] 2 consecutive failures in step openjdk23_checkpart2_java-matrix
- [8.x] 2 consecutive failures in pipeline elasticsearch-pull-request
- [8.x] 14 failures in test testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval (10.9% fail rate in 129 executions)
- [8.x] 3 failures in step part2 (13.0% fail rate in 23 executions)
- [8.x] 8 failures in step part-2 (20.5% fail rate in 39 executions)
- [8.x] 2 failures in step openjdk23_checkpart2_java-matrix (100.0% fail rate in 2 executions)
- [8.x] 3 failures in pipeline elasticsearch-intake (13.0% fail rate in 23 executions)
- [8.x] 8 failures in pipeline elasticsearch-pull-request (24.2% fail rate in 33 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -258,21 +258,9 @@ tests:
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry)}
   issue: https://github.com/elastic/elasticsearch/issues/115231
-- class: org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests
-  method: testAddWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval
-  issue: https://github.com/elastic/elasticsearch/issues/115339
-- class: org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests
-  method: testWatchWithLastCheckedTimeExecutesBeforeInitialInterval
-  issue: https://github.com/elastic/elasticsearch/issues/115354
-- class: org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests
-  method: testAddWithLastCheckedTimeExecutesBeforeInitialInterval
-  issue: https://github.com/elastic/elasticsearch/issues/115356
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testInferDeploysDefaultE5
   issue: https://github.com/elastic/elasticsearch/issues/115361
-- class: org.elasticsearch.xpack.watcher.trigger.schedule.engine.TickerScheduleEngineTests
-  method: testWatchWithNoLastCheckedTimeButHasActivationTimeExecutesBeforeInitialInterval
-  issue: https://github.com/elastic/elasticsearch/issues/115368
 - class: org.elasticsearch.reservedstate.service.FileSettingsServiceTests
   method: testProcessFileChanges
   issue: https://github.com/elastic/elasticsearch/issues/115280

--- a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/trigger/schedule/engine/TickerScheduleEngineTests.java
+++ b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/trigger/schedule/engine/TickerScheduleEngineTests.java
@@ -312,14 +312,13 @@ public class TickerScheduleEngineTests extends ESTestCase {
 
         engine.register(events -> {
             for (TriggerEvent ignored : events) {
-                if (runCount.get() == 0) {
+                if (runCount.getAndIncrement() == 0) {
                     logger.info("job first fire");
                     firstLatch.countDown();
                 } else {
                     logger.info("job second fire");
                     secondLatch.countDown();
                 }
-                runCount.incrementAndGet();
             }
         });
 
@@ -375,14 +374,13 @@ public class TickerScheduleEngineTests extends ESTestCase {
 
         engine.register(events -> {
             for (TriggerEvent ignored : events) {
-                if (runCount.get() == 0) {
+                if (runCount.getAndIncrement() == 0) {
                     logger.info("job first fire");
                     firstLatch.countDown();
                 } else {
                     logger.info("job second fire");
                     secondLatch.countDown();
                 }
-                runCount.incrementAndGet();
             }
         });
 
@@ -428,14 +426,13 @@ public class TickerScheduleEngineTests extends ESTestCase {
 
         engine.register(events -> {
             for (TriggerEvent ignored : events) {
-                if (runCount.get() == 0) {
+                if (runCount.getAndIncrement() == 0) {
                     logger.info("job first fire");
                     firstLatch.countDown();
                 } else {
                     logger.info("job second fire");
                     secondLatch.countDown();
                 }
-                runCount.incrementAndGet();
             }
         });
 
@@ -492,14 +489,13 @@ public class TickerScheduleEngineTests extends ESTestCase {
 
         engine.register(events -> {
             for (TriggerEvent ignored : events) {
-                if (runCount.get() == 0) {
+                if (runCount.getAndIncrement() == 0) {
                     logger.info("job first fire");
                     firstLatch.countDown();
                 } else {
                     logger.info("job second fire");
                     secondLatch.countDown();
                 }
-                runCount.incrementAndGet();
             }
         });
 

